napari/utils/colormaps/colormap.py

```diff
@@ -177,43 +176,15 @@ def map(self, values) -> np.ndarray:
         """
         values = np.atleast_1d(values)
 
-        precast = _cast_labels_to_minimum_dtype_auto(
-            values, len(self.colors) - 1, self.background_value
-        )
-
-        return self._map_precast(precast)
-
-    def _map_precast(self, values) -> np.ndarray:
-        """Map *precast* values to colors.
-
-        When mapping values, we first convert them to a smaller dtype for
-        performance reasons. This conversion changes the label values,
-        even for small labels. This method is used to map values that have
-        already been converted to the smaller dtype.
-
-        Parameters
-        ----------
-        values : np.ndarray
-            Values to be mapped. They must have already been downcast using
-            `_cast_labels_to_minimum_dtype_auto`.
+        if values.dtype.kind == 'f':
+            values = values.astype(np.int64)
 
-        Returns
-        -------
-        np.ndarray of shape (N, M, 4)
-            Mapped colors.
-        """
-        mapped = self.colors[values.astype(np.int64)]
-
-        mapped[values == self.background_value] = 0
-
-        # If using selected, disable all others
-        if self.use_selection:
-            cast_selection = _cast_labels_to_minimum_dtype_auto(
-                np.array([self.selection]),
-                len(self.colors) - 1,
-                self.background_value,
-            )[0]
-            mapped[values != cast_selection] = 0
+        mapped = self.colors[(values - 1) % (len(self.colors) - 1) + 1]
```

Comment:
Why not

```suggestion
        mapped = self.colors[_zero_preserving_modulo(values, len(self_colors), values.dtype)]
```

?

And probably that implies changes below... But I'm uncomfortable that this implementation is either duplicated or subtly different from that function...